pywt/_extensions/c/wavelets.c

```diff
@@ -328,36 +519,35 @@ Wavelet* blank_wavelet(size_t filters_length)
 {
     Wavelet* w;
 
-    if(filters_length < 1)
-        return NULL;
 
     /* pad to even length */
-    if(filters_length % 2)
+    if(filters_length > 0 && filters_length % 2)
         ++filters_length;
 
     w = wtmalloc(sizeof(Wavelet));
     if(w == NULL) return NULL;
 
     w->dec_len = w->rec_len = filters_length;
-
-    w->dec_lo_float = wtcalloc(filters_length, sizeof(float));
-    w->dec_hi_float = wtcalloc(filters_length, sizeof(float));
-    w->rec_lo_float = wtcalloc(filters_length, sizeof(float));
-    w->rec_hi_float = wtcalloc(filters_length, sizeof(float));
-
-    w->dec_lo_double = wtcalloc(filters_length, sizeof(double));
-    w->dec_hi_double = wtcalloc(filters_length, sizeof(double));
-    w->rec_lo_double = wtcalloc(filters_length, sizeof(double));
-    w->rec_hi_double = wtcalloc(filters_length, sizeof(double));
-
-    if(w->dec_lo_float == NULL || w->dec_hi_float == NULL ||
-       w->rec_lo_float == NULL || w->rec_hi_float == NULL ||
-       w->dec_lo_double == NULL || w->dec_hi_double == NULL ||
-       w->rec_lo_double == NULL || w->rec_hi_double == NULL){
-        free_wavelet(w);
-        return NULL;
+    if (filters_length > 0)
```

Comment:
This does not set the filter pointers to `NULL` if `filters_length == 0`. That should be done, or the allocation can be removed from the `if` statement (`malloc(0)` is allowed, but may return `NULL` so we should not check it afterwards).